### ISSUE TITLE
Documentation hole trying to configure gpg encrypted pillar data

### ISSUE THREAD

**Issue by ajfisher** [saltstack/salt#34126]
### Description of Issue/Question

The current docs have a hole in them regarding the configuration of the gnu-agent in order to be able to use a passphrase on the private keyfile which is the recommended approach here:

http://intothesaltmine.readthedocs.io/en/latest/chapters/secret-storage/architecture.html

Pertinent line:

```
You’ll need to manually launch the gpg-agent and then tell SaltStack where it can find the running agent. This
```

Basically I'm following this guide directly and whilst I can get the salt-master to decrypt the details in the pillar (as it obviously has access to the key locally), this won't work on the minion when executing `salt-call pillar.item secret` - in this case all I get is the encrypted information.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

It's all per directions - the only step I'm missing is how to get the agent to fire up on the request from the minion to unlock the file in order for the master to send the decrypted content.

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Follow set up directions from official docs: http://intothesaltmine.readthedocs.io/en/latest/chapters/secret-storage/introduction.html

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

```
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 18:00:18)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: Not Installed
               Tornado: Not Installed
 Debian source package: 2015.5.3+ds-1trusty1
```

**Comment by Ch3LL** [comment]
@ajfisher I tested this and it looks like this is fixed in 2015.8.3 and above. I am able to run `salt-call pillar.items` and see the gpg encrypted pillar data on the minion where I cannot in the 2015.5 branch. Can you try this on a version 2015.8.3 or above and see if it is working for you then?

**Comment by ajfisher** [comment]
Thanks @Ch3LL so the challenge I have is trying to figure out how to upgrade my master and minion to a later version of salt. After pretty much an entire afternoon of attempting to do this I've basically managed to nuke my salt master and can't go backwards or forwards at this point so... yay.

So I guess maybe if someone else can verify this is working in a later version then we can close this ticket off if it's expected not to work in 2015.5.3 and I'll muddle through trying to recover my master back to something that works again.

**Comment by Ch3LL** [comment]
@ajfisher i'm sorry you ran into that issue. Was there an error you were running into when attempting to upgrade that we can try to help you out with?

Also for reference [here](https://docs.saltstack.com/en/latest/faq.html#linux-unix) is some docs on an approach to upgrading your minions. Also we now host our own packages at http://repo.saltstack.com if that helps you out.

**Comment by ajfisher** [comment]
The main issue was to do with dealing with python-tornado and what ships by default in Ubuntu and there being no packages for 32 bit Ubuntu and I understand that SaltStack isn't going to build for 32-bit targets anyway.

Regardless I've rolled back and will try this at another point when I've upgraded servers etc to 64-bit machines and can effect an upgrade. 

Given this, I am going to close this out as the fix is that this isn't supported in 2015.5.3 which is cool.
